blocksync/reactor_test.go

```diff
@@ -177,12 +189,39 @@ func newReactor(
 		}
 	}
 
-	bcReactor := NewByzantineReactor(incorrectBlock, NewReactor(state.Copy(), blockExec, blockStore, blockSync, NopMetrics(), 0))
+	bcReactor := NewByzantineReactor(incorrectBlock, NewReactor(state.Copy(), blockExec, blockStore, blockSync, NopMetrics(), 0, ReactorVerifyData(verifyData)))
 	bcReactor.SetLogger(logger.With("module", "blocksync"))
 
 	return ReactorPair{bcReactor, proxyApp}
 }
 
+func TestVerifyData(t *testing.T) {
+	config = test.ResetTestRoot("blocksync_verify_data_test")
+	defer os.RemoveAll(config.RootDir)
+	genDoc, privVals := randGenesisDoc(1, false, 30)
+
+	maxBlockHeight := int64(10)
+
+	// Case 1: verifyData = true. Normal behavior.
+	// We cannot easily mock ProcessProposal failure here because we are using a real app (NewBaseApplication) via proxy.
+	// But we can observe that it works.
+	// To test failure, we would need to mock the app or the proxy connection.
+	// The current newReactor setup uses a real BaseApplication.
+	// So, passing verifyData=false should also work for normal blocks.
+	// To really test this, we might need a way to make ProcessProposal fail.
```

Comment:
I think it's fine not to test the failure of process proposal given it's a trivial change